homeassistant/components/freebox/router.py

```diff
@@ -89,7 +92,24 @@ async def update_all(self, now: datetime | None = None) -> None:
     async def update_device_trackers(self) -> None:
         """Update Freebox devices."""
         new_device = False
-        fbx_devices: list[dict[str, Any]] = await self._api.lan.get_hosts_list()
+
+        # Access to host list not available in bridge mode, API return error_code 'nodev'
+        try:
+            fbx_devices: list[dict[str, Any]] = await self._api.lan.get_hosts_list()
+        except HttpRequestError as err:
+            m = re.search('Request failed \(APIResponse: (.+?)\)', str(err))
+            if m:
+                try:
+                    json_str = m.group(1)
+                    json_resp = json.loads(json_str)
+                    if "error_code" in json_resp and json_resp['error_code'] == "nodev":
+                        _LOGGER.info("Host list is not available using bridge mode (%s)", json_resp['msg'])
+                        fbx_devices: list[dict[str, Any]] = []
+                except ValueError as ve:
+                    _LOGGER.error("Failed to parse JSON %s, error %s", json_str, ve)
```

Comment:
I'm wondering why making such a complicated code only to initialize a var for a specific error code.

Why not something like this ?
```suggestion
            if '"error_code": "nodev"' in str(err):
                fbx_devices: list[dict[str, Any]] = []
```

Until the library add an `error_code` attr to its errors ?

Reply:
it will also fix `black` formatting and `ruff`